src/plugins/intel_gpu/src/graph/layout_optimizer.cpp

```diff
@@ -1653,6 +1653,20 @@ format layout_optimizer::get_preferred_format(program_node& node) {
     auto output_layout = node.get_output_layout();
     bool use_onednn_impls = _optimization_attributes.use_onednn_impls;
 
+    bool allow_new_shape_infer = node.get_program().get_options().get<build_option_type::allow_new_shape_infer>()->enabled();
+
+    if (allow_new_shape_infer){
```

Comment:
```suggestion
    if (allow_new_shape_infer) {
```

Reply:
Currently the memory is not read and copied by read_vector, but it is just passing the pointer to the ngraph shape infer. So I think reading vector itself will impact the perf more, though there is no final perf data for now regarding the models using blocked formats :D Because so far only bert models with plain formats are investigated seriously. However we can consider many options in the future of course. Thanks for the review!